modules/4337/contracts/test/SafeMock.sol

```diff
@@ -216,8 +216,15 @@ contract Safe4337Mock is SafeMock, IAccount {
      */
     function _validateSignatures(PackedUserOperation calldata userOp) internal view returns (uint256 validationData) {
         (bytes memory operationData, uint48 validAfter, uint48 validUntil, bytes calldata signatures) = _getSafeOp(userOp);
-        checkSignatures(keccak256(operationData), operationData, signatures);
-        validationData = _packValidationData(false, validUntil, validAfter);
+
+        bytes32 dataHash = keccak256(operationData);
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+        (v, r, s) = _signatureSplit(signatures);
+        bool validSignature = owner == ecrecover(keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", dataHash)), v - 4, r, s);
+
+        validationData = _packValidationData(!validSignature, validUntil, validAfter);
```

Comment:
**Attention**

I had to inline the signature validation to Safe4337Mock because the `checkSignatures` function always reverted, which doesn't comply with the ERC4337 standard. It also made the simulations fail because if you use a placeholder signature, it'd just revert. Try/catch is not available for internal function calls.

I did not find any security implications, but I didn't spend much time on it since it's a mock contract.